### ISSUE TITLE
Table of contents: show first few automatically and click "show all" to see the whole list

### ISSUE THREAD

**Issue by cellio** [codidact/qpixel#289]
The new table of contents (thanks!) shows a TOC button, and if the user clicks on it the entire TOC is shown.  It's behind a button because we don't want to automatically show a huge TOC.  But I think it would be friendlier to show the top few entries automatically and then offer a "show all" button.  Currently, to see any answers you have to see all of them, but often what the user wants to see is just the top few.

In other words:

- Always show first N TOC entries (we can reuse the threshold that's already in there).
- Show button below that list if there are more.
- Button press replaces that with the complete TOC.  (Maybe someday we talk about pagination but that can be a problem for another day.)
